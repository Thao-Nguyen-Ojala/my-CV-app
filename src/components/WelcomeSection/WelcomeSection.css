video {
  object-fit: cover;
  width: 100% ;
  height: 100%;
  position: fixed;
  z-index: -1;
}

.welcome-container {
  /*background: url('../../media/images/theLostIsland.jpg') center center/cover no-repeat;*/
  height: 100vh;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  box-shadow: inset 0 0 0 100rem rgba(0,0,0,0.2);
  object-fit: contain;
}

.welcome-container > h1 {
  color: #fff;
  font-size: 8rem;
  margin: 20rem 0 0 0;
  text-align: center;
}

.welcome-container > p {
  margin-top: 5rem;
  margin-bottom: 2rem;
  color: #fff;
  font-size: 3.2rem;
  font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
}

.welcome-btns {
  margin-top: 3.2rem;
  display: flex;
  align-items: center;
  justify-content: center;
}

.welcome-btns .btn {
  margin: 6px;
}

.fa-play-circle {
  margin-left: 4px;
}

@media screen and (max-width: 960px) {
  .welcome-container > h1 {
    font-size: 6rem;
    margin-top: 15rem;
    text-align: center;
  }
}

@media screen and (max-width: 768px) {
  .welcome-container > h1 {
    font-size: 3.2rem;
    margin: 15rem 1rem 0 1rem;
    text-align: center;
  }

  .welcome-container > p {
    margin-bottom: 1rem;
    font-size: 2rem;
    text-align: center;
  }

  .welcome-btns {
    display: block; 
    align-items: stretch;
    justify-content: center;
  }

  .btn {
    width: 100%;
  }
}

@media screen and (max-width: 376px) {
  .welcome-container > h1 {
    margin-top: 5rem;
    font-size: 2.5rem;
  }

  .welcome-container > p {
    margin-bottom: 1rem;
    font-size: 2rem;
  }
  .welcome-btns {
    display: block; 
    align-items: stretch;
    justify-content: center;
  }
}

