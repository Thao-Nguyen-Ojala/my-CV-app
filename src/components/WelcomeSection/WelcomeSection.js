import React from 'react'
import './WelcomeSection.css'
import video1 from '../../media/videos/video-1.mp4'
import { Button } from '..'
import { Link as ScrollLink } from 'react-scroll';

import '../../App.css'

function WelcomeSection() {
  return (
    <div className='welcome-container'>
      <video src={video1} autoPlay loop muted={true} />
        <h1>Hello there, I am Thao!</h1>
        <p>Welcome to my page</p>
        <div className='welcome-btns'>
          <ScrollLink to='introduction' spy={true} smooth={true} duration={100}>
            <Button className='btns' buttonStyle='btn--outline' buttonSize='btn--large'>
              INTRODUCTION
            </Button>
          </ScrollLink>
          
          <ScrollLink to='skills' spy={true} smooth={true} duration={1000}>
            <Button className='btns' buttonStyle='btn--outline' buttonSize='btn--large'>
              SKILLS
            </Button>
          </ScrollLink>
          
          <ScrollLink to='experience' spy={true} smooth={true} duration={1000}>
            <Button className='btns' buttonStyle='btn--outline' buttonSize='btn--large'>
              EXPERIENCE
            </Button>
          </ScrollLink>
          
          <ScrollLink to='education' spy={true} smooth={true} duration={1000}>
            <Button className='btns' buttonStyle='btn--outline' buttonSize='btn--large' >
              EDUCATION
            </Button>
          </ScrollLink>
          
        </div>      
    </div>
  )
}

export default WelcomeSection
