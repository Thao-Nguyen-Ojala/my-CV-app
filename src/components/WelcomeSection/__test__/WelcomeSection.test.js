import React from 'react';
import ReactDOM from 'react-dom';
import WelcomeSection from '../WelcomeSection';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

afterEach(cleanup)

Object.defineProperty(HTMLMediaElement.prototype, 'muted', {
  set: () =>{}
})

it('render WelcomeSection without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <WelcomeSection>WelcomeSection testing</WelcomeSection>
    , div)
})

it('matches WelcomSection snapshot', () => {
  const testingWelcomeSection = renderer.create(<WelcomeSection></WelcomeSection>).toJSON();
  expect(testingWelcomeSection).toMatchSnapshot()
})