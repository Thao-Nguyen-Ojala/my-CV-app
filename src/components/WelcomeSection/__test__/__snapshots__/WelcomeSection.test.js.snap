// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches WelcomSection snapshot 1`] = `
<div
  className="welcome-container"
>
  <video
    autoPlay={true}
    loop={true}
    muted={true}
    src="video-1.mp4"
  />
  <h1>
    Hello there, I am Thao!
  </h1>
  <p>
    Welcome to my page
  </p>
  <div
    className="welcome-btns"
  >
    <a
      onClick={[Function]}
    >
      <button
        className="btn btn--outline btn--large null"
        data-testid="button"
      >
        INTRODUCTION
      </button>
    </a>
    <a
      onClick={[Function]}
    >
      <button
        className="btn btn--outline btn--large null"
        data-testid="button"
      >
        SKILLS
      </button>
    </a>
    <a
      onClick={[Function]}
    >
      <button
        className="btn btn--outline btn--large null"
        data-testid="button"
      >
        EXPERIENCE
      </button>
    </a>
    <a
      onClick={[Function]}
    >
      <button
        className="btn btn--outline btn--large null"
        data-testid="button"
      >
        EDUCATION
      </button>
    </a>
  </div>
</div>
`;
